CMakeLists.txt

```diff
@@ -90,14 +90,27 @@ if ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "GNU" OR
 
     set(CMAKE_C_FLAGS_DEBUG "${CMAKE_C_FLAGS_DEBUG} -Og")
     set(CMAKE_CXX_FLAGS_DEBUG "${CMAKE_CXX_FLAGS_DEBUG} -Og")
+endif()
 
-    # Add -march=${TYPE_ARCHITECTURE} (-march=native by default) to the release
-    # build flags
+# Add -march=${TYPE_ARCHITECTURE} (-march=native by default) to the release
+# build flags
+if("${CMAKE_CXX_COMPILER_ID}" STREQUAL "GNU")
+  set(CMAKE_C_FLAGS_RELEASE "${CMAKE_C_FLAGS_RELEASE} -march=${TYPE_ARCHITECTURE}")
+  set(CMAKE_CXX_FLAGS_RELEASE "${CMAKE_CXX_FLAGS_RELEASE} -march=${TYPE_ARCHITECTURE}")
+  set(CMAKE_C_FLAGS_RELWITHDEBINFO "${CMAKE_C_FLAGS_RELWITHDEBINFO} -march=${TYPE_ARCHITECTURE}")
+    set(CMAKE_CXX_FLAGS_RELWITHDEBINFO "${CMAKE_CXX_FLAGS_RELWITHDEBINFO} -march=${TYPE_ARCHITECTURE}")
+elseif("${CMAKE_CXX_COMPILER_ID}" STREQUAL "Clang" OR
+  "${CMAKE_CXX_COMPILER_ID}" STREQUAL "AppleClang")
+  # detect if the local architechture is arm64 or x86_64
+  if(CMAKE_HOST_SYSTEM_PROCESSOR STREQUAL "arm64")
+    set(CMAKE_OSX_ARCHITECTURES "x86_64;arm64" CACHE STRING "")
```

Comment:
Why build a universal binary instead of detecting which architecture we actually need?

Reply:
I thought it would be needed to handle the case where you have a python distribution built with `x86_64` while working on an `arm64`. I need @rosecers input to check that it's actually working though.

Reply:
My understanding is that if Python is an `x86_64` binary, it will only need `x86_64` libraries. Rosetta 2 will then deal with the emulation of x86 on ARM. The main issue with building universal binary is the increased compile time (and file size when we upload this to PyPI). 

If there is an easy way to detect the required architecture insead of doing both that would be nice, otherwise we'll survive.

Reply:
I could not find one.